c++/src/kj/compat/http.c++

```diff
@@ -1943,7 +1943,7 @@ public:
     size_t alreadyRead = 0;
 
     for (;;) {
-      if (length == 0) {
+      if (length == alreadyRead) {
```

Comment:
The delta here could likely be minimized by just adding an additional size_t field for the full length as opposed to changing the way `length` is calculated/managed now. But consider that optional. LGTM so long as ci looks good

Reply:
No, please do not change the meaning of `length` like this. This logic is delicate and it's just too easy to introduce bugs. For instance, as of right now this PR introduces a severe HTTP desync bug because line 1953 has not been updated for the new meaning of `lengt`h.

Reply:
oh doh, I'd actually spotted that right when starting the review, managed to get distracted, then forgot about it by the time I actually wrote the review. Good catch.